### PR TITLE
Modified MG pod tolerations

### PR DESCRIPTION
**Pull request by apoorvajagtap**
Fixes: https://github.com/openshift/oc/issues/941
As per the current tolerations, `Operator: Exists`, the must-gather pod is spawned with following `spec.tolerations`:

> $ oc get pods must-gather-2cnh8 -ojson -n openshift-must-gather-strbd | jq .spec.tolerations
> [
>   {
>     "operator": "Exists"
>   }
> ]

After specifying the `Key: "node-role.kubernetes.io/master"`, the pod comes up with these additional tolerations (which helps to take the `not-ready` and `unreachable` conditions into consideration):

> $ oc get pods must-gather-lcq28 -ojson -n openshift-must-gather-j245c | jq .spec.tolerations
> [
>   {
>     "key": "node-role.kubernetes.io/master",
>     "operator": "Exists"
>   },
>   {
>     "effect": "NoExecute",
>     "key": "node.kubernetes.io/not-ready",
>     "operator": "Exists",
>     "tolerationSeconds": 300
>   },
>   {
>     "effect": "NoExecute",
>     "key": "node.kubernetes.io/unreachable",
>     "operator": "Exists",
>     "tolerationSeconds": 300
>   }
> ]

### CODE REVIEW

**Comment by apoorvajagtap** [comment]
Hi @deads2k @soltysh, I would really like to know your inputs on this one, whenever possible.
In case I have missed any specific part of the workflow, please let me know and I'll abide by that. Thank you!

File: pkg/cli/admin/mustgather/mustgather.go [hunk 1]
```diff
@@ -649,6 +649,7 @@ func (o *MustGatherOptions) newPod(node, image string) *corev1.Pod {
 			TerminationGracePeriodSeconds: &zero,
 			Tolerations: []corev1.Toleration{
 				{
+					Key:      "node-role.kubernetes.io/master",
 					Operator: "Exists",
```
Comment (soltysh) [inline pkg/cli/admin/mustgather/mustgather.go:653]: This particular use-case is to ensure that we match all possible taints, see Note section in https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/ I'd suggest adding another toleration like the one you're proposing, but only when node wasn't specified. Your current code will break that last assumption.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/cli/admin/mustgather/mustgather.go (hunk 1 above)
Comment (apoorvajagtap) [reply pkg/cli/admin/mustgather/mustgather.go:657]: Thank you @soltysh. I have modified the toleration considering the conditions. Let me know if this fits better or not.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by soltysh** [comment]
This PR is being reverted in #1003 since it's blocking k8s bump, also upon further investigation I see the current approach with tolerations as intended and we won't change it.

**Pull request merged**